### DISCUSSION TITLE
有办法获取消息开头为“@某人”的信息吗

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#222] [Q&A] [answered]
是我个人的需求，希望在检测到`@机器人`的时候作出回应。

但是目前感觉机器人不能获取开头为`@某人`的消息，结构是这样的：
```
// 这条消息不能够通过/.*/g捕获
recv from: [Group: ……] [CQ:at,qq=…,text=@机器人] test
```
不过如果@消息在中间的话就可以正常获取。
```
// 这条消息能够通过/.*/g捕获
recv from: [Group: ……] test [CQ:at,qq=…,text=@机器人] test
```

所以想请教一下有没有办法捕获并处理第一种信息，也就是`@机器人 消息正文`的情况

**Comment by Arondight** [comment]
我正在摸鱼的时间做，等一等吧，最迟一天 #210

**Comment by Arondight** [comment, marked as answer]
功能我基本上做好了，我多问一下，当处理完 "@bot command" 之后，我在 raw_message 中删除了 "@bot "，message 中会有一个 "[{ type: 'at', data: { qq: 11, text: '@bot' } }]"，我测试的时候顺手给过滤掉了，你希望保留么？

更进一步说，如果你希望保留，你要保留完整的数据，还是希望我额外在 message 中增加一个真值，表示这条消息是不是有过 "@bot" 呢

**Reply by Arondight** [reply]
如果你只想知道这条消息是不是圈了机器人，那么就没必要保留原始数据结构，你从原始数据结构中判断出这个结论还是要写代码的。如果你根本不关心是不是圈了机器人，只希望机器人响应圈他的信息，那么真值都没必要保留

**Reply by mark9804** [reply]
我现在在处理事情，晚上六七点的时候仔细回复吧

目前的想法是只要说明文档能写清楚会返回一个什么样的结构，那么从可扩展性角度来说的话信息越详细越好

**Comment by mark9804** [comment]
刚刚想了一下。

我觉得需求有两个，按优先级排列的话是：

1. 最重要，能够获取@bot的消息；
2. 能够知道这条消息@过bot。

其中第二点只要用true/false说明应该就可以了，即使是对于人来说被人@了本身也没有什么丰富的信息量，只是有一个"这条消息是找我的"的概念。对于机器人也一样，true/false就够用了。

**Reply by Arondight** [reply]
我已经把提交推上去了，文档参见 https://github.com/Arondight/Adachi-BOT/blob/ca52feb4a80998cd048313d76c72e33382d1997f/docs/%E5%BC%80%E5%8F%91%E6%8C%87%E5%BC%95.md#oicq-%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%BF%AE%E6%94%B9

**Reply by Arondight** [reply]
提交 https://github.com/Arondight/Adachi-BOT/commit/ca52feb4a80998cd048313d76c72e33382d1997f

**Reply by mark9804** [reply]
> 提交 https://github.com/Arondight/Adachi-BOT/commit/ca52feb4a80998cd048313d76c72e33382d1997f

感谢，我回去看看
